mixins/harvester-vm/index.js

```diff
@@ -858,8 +858,10 @@ export default {
       const _QGA_JSON = this.getMatchQGA(osType);
 
       userDataJson.package_update = true;
-      if (Array.isArray(userDataJson.packages) && !userDataJson.packages.includes('qemu-guest-agent')) {
-        userDataJson.packages.push('qemu-guest-agent');
+      if (Array.isArray(userDataJson.packages)) {
+        if (!userDataJson.packages.includes('qemu-guest-agent')) {
+          userDataJson.packages.push('qemu-guest-agent');
+        }
       } else {
         userDataJson.packages = QGA_JSON.packages;
       }
```

Comment:
Tiny option: As `Array.isArray(userDataJson.packages)` do noting, I think the code can change like follow logic:

```suggestion
      if (!userDataJson.packages) {
        userDataJson.packages = QGA_JSON.packages;
      } else if (Array.isArray(userDataJson.packages) && !userDataJson.packages.includes('qemu-guest-agent')) {
         userDataJson.packages.push('qemu-guest-agent');
      }
```

Reply:
Like we just discussed, it might be a good choice to keep it that way.